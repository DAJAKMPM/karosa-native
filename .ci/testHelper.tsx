import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
