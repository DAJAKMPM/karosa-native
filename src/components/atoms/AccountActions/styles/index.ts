/**
 *
 * @format
 *
 */

import { StyleSheet } from "react-native";
import { theme } from "@app/styles";

export const styles = StyleSheet.create({
  actionContainer: { flexDirection: "column", flex: 1, alignItems: "center" },
  container: {
    flex: 1,
    justifyContent: "flex-start",
    flexDirection: "row",
    paddingLeft: 8,
    paddingRight: 8,
  },
  label: {
    ...theme.textLight,
    fontWeight: "700",
    marginBottom: 12,
    position: "absolute",
    bottom: 5,
  },
});
