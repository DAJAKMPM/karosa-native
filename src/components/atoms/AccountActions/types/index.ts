/**
 *
 * @format
 *
 */

type Actions = {
  label: string;
  icon: React.ReactElement;
  onPress: () => void;
};

export type Props = {
  actions: Actions[];
};
