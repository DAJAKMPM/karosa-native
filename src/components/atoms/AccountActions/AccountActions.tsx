/**
 *
 * AccountActions
 * @format
 *
 */

import React, { FC } from "react";
import { View, TouchableOpacity } from "react-native";
import Text from "@app/atoms/Text";

import type { Props } from "./types";
import { styles } from "./styles";

const AccountActions: FC<Props> = (props) => {
  const { actions } = props;

  return (
    <View style={styles.container}>
      {actions.map((action, key) => (
        <TouchableOpacity
          key={key}
          style={styles.actionContainer}
          onPress={action.onPress}
        >
          {action.icon}
          <Text textStyle={styles.label} text={action.label} />
        </TouchableOpacity>
      ))}
    </View>
  );
};

export default AccountActions;
