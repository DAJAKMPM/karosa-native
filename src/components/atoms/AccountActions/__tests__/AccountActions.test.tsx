import React from "react";
import { noop } from "lodash";
import { EvilIcons } from "@expo/vector-icons";
import { renderWithWrapper } from "@app/ci/testHelper";
import AccountActions from "../AccountActions";

describe("AccountActions Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <AccountActions
        actions={[
          {
            icon: <EvilIcons name="credit-card" />,
            label: "To Pay",
            onPress: noop,
          },
        ]}
      />
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
