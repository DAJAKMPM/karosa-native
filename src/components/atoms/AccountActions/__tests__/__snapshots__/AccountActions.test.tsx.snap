// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountActions Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingLeft": 8,
        "paddingRight": 8,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "opacity": 1,
        }
      }
    >
      <Text />
      <Text
        style={
          Object {
            "bottom": 5,
            "color": "#1C1C1C",
            "fontFamily": "SFProText-Regular",
            "fontSize": 12,
            "fontWeight": "700",
            "lineHeight": 14.06,
            "marginBottom": 12,
            "position": "absolute",
          }
        }
      >
        To Pay
      </Text>
    </View>
  </View>
</View>
`;
